### PR TITLE
add docs for model parameters in prompt registry

### PR DESCRIPTION
**Pull request by abubakarsohail**

### CODE REVIEW

File: openapi.json [hunk 1]
```diff
@@ -378,6 +378,28 @@
...
+                "title": "Model Provider",
+                "description": "The model provider of the prompt",
+                "enum": ["openai", "anthropic"]
+              },
+              "model": {
+                "type": "string",
+                "title": "Model",
+                "description": "The model of the prompt"
+              },
+              "parameters": {
```
Comment (jzone3) [inline openapi.json:393]: I'm worried this won't be kept up-to-date. OpenAI changes frequently and so does Anthropic. Can we just have parameters set to any? We can link to their docs instead.

File: images/model-parameters.png
Comment (jzone3) [inline images/model-parameters.png:1]: Why does it say "open"?

File: features/prompt-registry.mdx [hunk 2]
```diff
@@ -121,12 +129,36 @@ promptlayer.prompts.publish(
...
+
+They are also available in the following REST API endpoints
 
 `/rest/get-prompt-template` ([read more](/reference/get-prompt-template))
 
 `/rest/publish-prompt-template` ([read more](/reference/publish-prompt-template))
 
+### Model Parameters
+
+You can also add model parameters to your prompt templates. This allows you to set the provider, model, temperature, max_tokens, and other parameters for each prompt template. This is useful if you want to use different parameters for different prompts.
```
Comment (jzone3) [inline features/prompt-registry.mdx:141]: Should be: 

`provider`, `model`, `temperature`... etc

Comment (jzone3) [inline features/prompt-registry.mdx:141]: Remove "This is useful if you want to use different parameters for different prompts."

Comment (jzone3) [inline features/prompt-registry.mdx:141]: "You can also add model parameters to your prompt templates."

-->

"Custom model parameters can be associated with individual prompt templates versions.

File: openapi.json (hunk 1 above)
Comment (jzone3) [reply openapi.json:393]: In fact, are we type-checking this in the API? If so, we should remove it.

It's possible OpenAI has hidden kwargs that some customers know about and want to use. We should allow that.

File: features/prompt-registry.mdx [hunk 3]
```diff
@@ -51,6 +51,14 @@ You can also optionally pass `version` to get an older version of a prompt. By d
 template_dict = promptlayer.prompts.get("my_template", version=3)
 ```
 
+### Model Parameters
+
+When you fetch a prompt template, you can get the model parameters that were used by passing in the `include_parameters=True` argument. This returns a tuple with the prompt template and the model parameters. For example:
+
+```python Python SDK
+prompt, model = promptlayer.prompts.get("my_template", include_parameters=True)
```
Comment (jzone3) [inline features/prompt-registry.mdx:59]: "prompt, model =" -> "template, model_params ="

File: features/prompt-registry.mdx [hunk 4]
```diff
@@ -51,6 +51,14 @@ You can also optionally pass `version` to get an older version of a prompt. By d
 template_dict = promptlayer.prompts.get("my_template", version=3)
 ```
 
+### Model Parameters
+
+When you fetch a prompt template, you can get the model parameters that were used by passing in the `include_parameters=True` argument. This returns a tuple with the prompt template and the model parameters. For example:
```
Comment (jzone3) [inline features/prompt-registry.mdx:56]: Rephrase the paragraph to (also make sure the last sentence is correct):


When fetching a prompt template, you can optionally retrieve model parameters by using `include_parameters=True`. This will provide you with a tuple containing both the prompt template and the model parameters. 

If no custom model parameters are set, the value will be an empty dictionary.

**Review by jzone3 — CHANGES_REQUESTED** [review]

File: features/prompt-registry.mdx (hunk 4 above)
Comment (jzone3) [reply features/prompt-registry.mdx:56]: GPT helped me clean up make the wording sound a little cleaner

File: openapi.json (hunk 1 above)
Comment (abubakarsohail) [reply openapi.json:393]: @jzone3 do you mean to remove the existing validations as well or just allow any additional ones?

File: images/model-parameters.png
Comment (abubakarsohail) [reply images/model-parameters.png:1]: The new image
model-parameters

File: features/prompt-registry.mdx (hunk 2 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx]: @jzone3 updated

Comment (abubakarsohail) [reply features/prompt-registry.mdx]: @jzone3 updated.

Comment (abubakarsohail) [reply features/prompt-registry.mdx]: @jzone3 updated.

File: features/prompt-registry.mdx (hunk 3 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx]: @jzone3 updated.

File: features/prompt-registry.mdx (hunk 4 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx]: @jzone3 updated. Skipped last line because the dictionary will have attributes only with none values like
```py
{"model_provider": None, "model": None, "parameters": None}
```

File: openapi.json (hunk 1 above)
Comment (jzone3) [reply openapi.json:393]: I think remove. The maintenance cost is too high.

Comment (jzone3) [reply openapi.json:393]: Like parameters should just be <string, any>

**Review by jzone3 — CHANGES_REQUESTED** [review]
Please remove `parameters` validations. If you want to talk about this we can

File: openapi.json (hunk 1 above)
Comment (abubakarsohail) [reply openapi.json:392]: @jzone3 removed validations, this is how the API looks like now
Screenshot 2023-09-08 at 3 44 55 PM
Screenshot 2023-09-08 at 3 45 11 PM

**Comment by abubakarsohail** [comment]
@jzone3 updated docs after changing it to metadata

Screenshot 2023-09-12 at 2 01 49 PM
Screenshot 2023-09-12 at 2 02 16 PM
Screenshot 2023-09-12 at 2 02 38 PM
Screenshot 2023-09-12 at 2 03 12 PM
Screenshot 2023-09-12 at 2 03 28 PM

File: features/prompt-registry.mdx [hunk 5]
```diff
@@ -98,7 +106,7 @@ Prompt labels are a way to put a label on your prompt template to help you organ
 
 ### Commit Messages
 
-Prompt commit messages allow you to set a brief 72 characters-long description on each of your prompt template version to help you keep track of changes. 
+Prompt commit messages allow you to set a brief 72 characters-long description on each of your prompt template version to help you keep track of changes.
```
Comment (jzone3) [inline features/prompt-registry.mdx:109]: "character-long" not "characters-long"

File: features/prompt-registry.mdx [hunk 6]
```diff
@@ -121,12 +129,35 @@ promptlayer.prompts.publish(
     prompt_template=pt,
 )
 ```
-They are also available in the following REST API endpoints 
+
+They are also available in the following REST API endpoints
```
Comment (jzone3) [inline features/prompt-registry.mdx:133]: "available through" not "available in"

File: features/prompt-registry.mdx [hunk 7]
```diff
@@ -121,12 +129,35 @@ promptlayer.prompts.publish(
...
 
 `/rest/get-prompt-template` ([read more](/reference/get-prompt-template))
 
 `/rest/publish-prompt-template` ([read more](/reference/publish-prompt-template))
 
+### Metadata
+
+Custom metadata can be associated with individual prompt templates versions. This allows you to set the provider, model, temperature... etc for each prompt template. It is important to note here that `model` attribute is reserved for the model parameters.
```
Comment (jzone3) [inline features/prompt-registry.mdx:141]: "individual prompt template versions" not "individual prompt templates versions"

Comment (jzone3) [reply features/prompt-registry.mdx:141]: "This allows you to set values such as provider, model, or temperature for each prompt template."

Not

"This allows you to set the provider, model, temperature... etc for each prompt template."

File: features/prompt-registry.mdx [hunk 8]
```diff
@@ -121,12 +129,35 @@ promptlayer.prompts.publish(
...
+
+<img src="/images/model-parameters.png" />
+
+You can also use our python SDK to publish a prompt template with metadata. For example:
```
Comment (jzone3) [inline features/prompt-registry.mdx:145]: Python should be capitalized

File: openapi.json [hunk 9]
```diff
@@ -560,6 +562,41 @@
...
+        "properties": {
+          "provider": { "title": "Provider", "type": "string" },
+          "name": { "title": "Name", "type": "string" },
+          "parameters": {
+            "title": "Parameters",
+            "default": {},
+            "type": "object"
+          }
+        },
+        "required": ["provider", "name"]
```
Comment (jzone3) [inline openapi.json:578]: Why are these required? Can't they be null?

**Review by jzone3 — APPROVED** [review]
Looks good.

Please just correct all comments. Once every comment is resolved feel free to merge

File: openapi.json (hunk 9 above)
Comment (abubakarsohail) [reply openapi.json:578]: @jzone3 don't we always need a provider and name for the model?

Comment (jzone3) [reply openapi.json:578]: What if the user doesn't want to specify any metadata at all?

File: features/prompt-registry.mdx (hunk 5 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:109]: Updated

File: features/prompt-registry.mdx (hunk 6 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:133]: Updated

File: features/prompt-registry.mdx (hunk 7 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:141]: Updated

File: openapi.json (hunk 9 above)
Comment (abubakarsohail) [reply openapi.json:578]: @jzone3 the user can choose not to send metadata at all or send it as null.
